src/main/java/co/cask/hydrator/salesforce/plugin/source/batch/SalesforceSourceConfig.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.plugin.source.batch;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Macro;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.etl.api.validation.InvalidConfigPropertyException;
+import co.cask.hydrator.salesforce.SObjectDescriptor;
+import co.cask.hydrator.salesforce.parser.SOQLParsingException;
+import co.cask.hydrator.salesforce.parser.SalesforceQueryParser;
+import co.cask.hydrator.salesforce.plugin.BaseSalesforceConfig;
+import co.cask.hydrator.salesforce.plugin.source.batch.util.SalesforceQueryUtil;
+import co.cask.hydrator.salesforce.plugin.source.batch.util.SalesforceSourceConstants;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.sforce.ws.ConnectionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * This class {@link SalesforceSourceConfig} provides all the configuration required for
+ * configuring the {@link SalesforceBatchSource} plugin.
+ */
+public class SalesforceSourceConfig extends BaseSalesforceConfig {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SalesforceSourceConfig.class);
+
+  @Name(SalesforceSourceConstants.PROPERTY_QUERY)
+  @Description("The SOQL query to retrieve results from. Example: select Id, Name from Opportunity")
+  @Nullable
```

Comment:
Why are some of these nullable ? If user does not specify would we still be able to retrieve data from salesforce?

Reply:
Because in this PR we added query by SObject name option, so user can specify SOQL or SObject name to retrieve data from Salesforce. If both of them will be empty validation will fail.